src/core/EntrySearcher.cpp

```diff
@@ -182,11 +182,12 @@ bool EntrySearcher::searchEntryImpl(Entry* entry)
                     && term.regex.match(entry->attributes()->value(term.word)).hasMatch();
             break;
         default:
-            // Terms without a specific field try to match title, username, url, and notes
+            // Terms without a specific field try to match title, username, url, notes, and groups
             found = term.regex.match(entry->resolvePlaceholder(entry->title())).hasMatch()
                     || term.regex.match(entry->resolvePlaceholder(entry->username())).hasMatch()
                     || term.regex.match(entry->resolvePlaceholder(entry->url())).hasMatch()
-                    || term.regex.match(entry->notes()).hasMatch();
+                    || term.regex.match(entry->notes()).hasMatch()
+                    || !entry->group()->hierarchy(100).filter(term.regex).empty();
```

Comment:
This is extremely inefficient way to do this. You will be repeating string building and filtering for EACH entry. Since there are less groups than entries you should simply iterating the group list to search for each group name and adding all the entries in that group.

Reply:
sorry, I didn't understand.
It should be better now, but is it good enough?